.github/workflows/ci.yml

```diff
@@ -34,7 +34,7 @@ jobs:
           cmake -S test -B build/ \
           -G "Unix Makefiles" \
           -DCMAKE_BUILD_TYPE=Debug \
-          -DBUILD_UNIT_TESTS=ON \
+          -DUNITTEST=ON \
```

Comment:
If we are changing this, might as well change it to BUILD_TESTING which is the standard variable for enabling/disabling testing (see https://cmake.org/cmake/help/latest/module/CTest.html, though its commonly used for testing even if not using ctest).

Reply:
`BUILD_UNIT_TESTS` is also not a cmake standard variable. This macro is updated to align other FreeRTOS libraries. Since it is only used in FreeRTOS libraries, using `UNITTEST` doesn't seem to have problem.